hw/ip/i2c/doc/_index.md

```diff
@@ -1,3 +1,223 @@
+
+The I2C hardware interface consists of two external pins, SCL and SDA, whose behavior is described in the I2C specification.
+These pins are typically controlled by an internal state machine.
+However, there is a simpler “override” mode, by which these pins can be directly addressed by software.
+This override mode is useful for troubleshooting or error-recovery.
+
+To enter override mode, the register field {{< regref OVRD.TXOVRDEN >}} is asserted by software.
+In this state the output drivers scl_tx_o and sda_tx_o are controlled directly by the register fields {{< regref OVRD.SCLVAL >}} () and {{< regref OVRD.SDAVAL >}}.
+When {{< regref OVRD.SCLVAL >}} and {{< regref OVRD.SDAVAL >}} are set high, the virtual open drain configuration will leave the output resistively pulled high, and controllable by remote targets.
+In this state, with SCL or SDA asserted high, the register fields {{< regref VAL.SCL_RX >}} and {{< regref VAL.SDA_RX >}} can be used to receive inputs (including remote acknowledgments) from target devices.
+
+### Byte-Formatted Programming Mode 
+
+The state machine-controlled mode allows for higher-speed operation with less frequent software interaction.
+In this mode, the I2C pins are controlled by the I2C state machine, which in turn is controlled by a sequence of formatting indicators.
+The formatting indicators indicate:
+- The sequence of bytes which should be transmitted on the SDA and SCL pins.
+- The periods between transmitted bytes when the state-machine should stop
+transmission and instead read back a fixed number of bytes.
+- Which bytes should be preceded by a START symbol.
+- Which bytes should be followed by a STOP symbol
+The format indicator consists of 13-bits. That is of one single Format Byte (entered into the format FIFO through {{< regref FDATA.FBYTE >}}), and five (5) 1-bit flags (entered into the format FIFO through registers {{< regref FDATA.READ >}}, {{< regref FDATA.RCONT >}}, {{< regref FDATA.START >}}, {{< regref FDATA.STOP >}} and {{< regref FDATA.NAKOK >}})
+
+The I2C reads each format indicator from the head of FMT_FIFO, and processes them in turn.
+If none of the flags are set for the format format indicator none, the I2C FSM simply transmits the Format Byte onto the SCL and SDA pins according to the specification, waits for acknowledgement, and then proceeds to the next format indicator.
+The format flags modulate the behavior as follows.
+- READ (Corresponds to {{< regref FDATA.READ >}}):
+Signifies the Format Byte ({{< regref FDATA.FBYTE >}}) should be treated as an unsigned number, R, and prompts the state machine to read R bytes from the target device.
+Bytes read from the bus, are inserted into the RX FIFO where they can be accessed by software.
+A value of 0 is treated as a read of 256B.
+To read a larger byte stream, multiple 256B reads can be chained together using the RCONT flag.
+- RCONT (Corresponds to FIFO inputs {{< regref FDATA.RCONT >}}, Only used with READ):
```

Comment:
Corresponds -> corresponds
Only -> only

(same replacement should be done below)